subprojects/build-init/src/main/groovy/org/gradle/api/tasks/wrapper/Wrapper.java

```diff
@@ -63,6 +64,7 @@
     private PathBase distributionBase = PathBase.GRADLE_USER_HOME;
     private String distributionUrl;
     private GradleVersion gradleVersion;
+    private String distType;
```

Comment:
`DistributionLocator` is actually an internal class - it has no JavaDoc. As such it should not contain domain classes - like `DistributionType`. I would suggest haven `DistributionType` in `Wrapper`. If we would want to use `DistributionType` all over the place it should be somewhere in `org.gradle.api` in Gradle `core. But I think this would be too much for now.